src/vs/workbench/contrib/chat/browser/chatVariables.ts

```diff
@@ -25,23 +26,53 @@ interface IChatData {
 	resolver: IChatVariableResolver;
 }
 
+/**
+ * Helper to run provided `jobs` in parallel and return only
+ * the `successful` results in the same order as the original jobs.
+ */
+const getJobResults = async (
+	jobs: Promise<IChatRequestVariableEntry | null>[],
+): Promise<IChatRequestVariableEntry[]> => {
+	return (await Promise.allSettled(jobs))
+		// filter out `failed` and `empty` results
+		.filter((result) => {
+			return result.status !== 'rejected' && result.value !== null;
+		})
+		// extract the result value
+		.map((result) => {
+			// assertions below must always be true because of the filter logic above
+			assert(
+				result.status === 'fulfilled',
+				`Failed to resolve variables: unexpected promise result status "${result.status}".`,
+			);
+			assert(
+				result.value !== null,
+				`Failed to resolve variables: promise result must not be null.`,
+			);
+
+			return result.value;
+		});
+};
+
 export class ChatVariablesService implements IChatVariablesService {
 	declare _serviceBrand: undefined;
 
-	private _resolver = new Map<string, IChatData>();
+	private readonly _resolver = new Map<string, IChatData>();
 
 	constructor(
 		@IChatWidgetService private readonly chatWidgetService: IChatWidgetService,
 		@IViewsService private readonly viewsService: IViewsService,
-	) {
-	}
+	) { }
 
-	async resolveVariables(prompt: IParsedChatRequest, attachedContextVariables: IChatRequestVariableEntry[] | undefined, model: IChatModel, progress: (part: IChatVariableResolverProgress) => void, token: CancellationToken): Promise<IChatRequestVariableData> {
-		let resolvedVariables: IChatRequestVariableEntry[] = [];
-		const jobs: Promise<any>[] = [];
-
-		prompt.parts
-			.forEach((part, i) => {
+	public async resolveVariables(
```

Comment:
Mostly readability improvements in this file, the logic remains the same.

Reply:
Yeah sorry, it was very hard to read even on 27" monitor, so I was formatting it on the go as I was parsing through the code, and then decided to keep it. The same applies for other cases where I split long lines over. Let me know if you think that reverting this would help with the review at this point 👂

Reply:
I also recommend scoping PRs down where possible, large PRs with many changes make it hard on the reviewers and ultimately delay landing features.